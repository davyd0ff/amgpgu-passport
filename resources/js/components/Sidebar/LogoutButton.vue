<template>
  <a
    href="#"
    class="black-text exit-to-app"
    v-on:click.prevent="onLogout"
    title="LOGIN_LOGOUT"
  >
    <i class="material-icons">exit_to_app</i>
  </a>
</template>

<script>
export default {
  name: 'LogoutButton',
  methods: {
    onLogout: function () {
      this.$store.dispatch('logout').then(() => {
        // route to login
      });
    },
  },
};
</script>

<style scoped></style>
