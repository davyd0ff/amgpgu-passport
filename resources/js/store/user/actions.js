import PassportApi from '@/commands/passport';

export default {
  login: async ({ commit }, { username, password }) => {
    try {
      const response = await PassportApi.login({ username, password });
      commit('LOGIN', { token: response.data });
    } catch (error) {
      throw new Error(error);
    }
  },

  logout: async ({ commit }) => {
    try {
      await PassportApi.logout();
      commit('LOGOUT');
    } catch (error) {
      throw new Error(error);
    }
  },

  getUserData: async ({ commit }) => {
    try {
      const response = await PassportApi.getUserData();
      commit('SET_USER', { user: response.data });
    } catch (error) {
      throw new Error(error);
    }
  },

  getUserMenu: async ({ commit }) => {
    try {
      const response = await PassportApi.getUserMenu();
      commit('SET_MENU', { menu: response.data });
    } catch (error) {
      throw new Error(error);
    }
  },

  uploadAvatar: async ({ commit }, { file }) => {
    try {
      const response = await PassportApi.uploadAvatar(file);
      commit('SET_AVATAR', { url: response.data });
    } catch (error) {
      throw new Error(error);
    }
  },
};
