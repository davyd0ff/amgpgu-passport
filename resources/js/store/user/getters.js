export default {
  user: function (state) {
    return state.user;
  },
  avatar: function (state) {
    return state.user.avatar ?? '';
  },
  menu: function (state) {
    return state.user.menu;
  },
};
