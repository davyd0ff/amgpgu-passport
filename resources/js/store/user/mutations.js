export default {
  LOGIN: function (_, { token }) {
    // todo: save token
    window.localStorage.setItem('token', JSON.stringify(token));
  },
  LOGOUT: function (state) {
    // todo fixme: don't reset state ... wtf?
    state = {};
    window.localStorage.removeItem('token');
  },
  SET_USER: function (state, { user }) {
    state.user = user;
  },
  SET_AVATAR: function (state, { url }) {
    state.user = { ...state.user, avatar: url };
  },
  SET_MENU: function (state, { menu }) {
    state.user = { ...state.user, menu: menu };
  },
};
