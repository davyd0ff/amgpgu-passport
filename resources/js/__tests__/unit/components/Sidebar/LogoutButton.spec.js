import { shallowMount, createLocalVue } from '@vue/test-utils';
import LogoutButton from '@/components/Sidebar/LogoutButton.vue';
import Vuex from 'vuex';
import user from '@/store/user';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('LogoutButton.vue', () => {
  let store;
  const createWrapper = () => {
    return shallowMount(LogoutButton, { store, localVue });
  };

  beforeEach(() => {
    user.actions.logout = jest.fn();

    store = new Vuex.Store({
      modules: {
        user,
      },
    });
  });

  it('render', () => {
    const wrapper = createWrapper();

    expect(wrapper.element).toMatchSnapshot();
  });

  it('logout was called when to press on the button', () => {
    const wrapper = createWrapper();
    const logoutButton = wrapper.find('a');

    logoutButton.trigger('click');

    expect(user.actions.logout).toBeCalled();
  });
});
