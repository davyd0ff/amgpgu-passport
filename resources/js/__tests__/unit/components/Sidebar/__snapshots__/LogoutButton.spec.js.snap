// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutButton.vue render 1`] = `
<a
  class="black-text exit-to-app"
  href="#"
  title="LOGIN_LOGOUT"
>
  <i
    class="material-icons"
  >
    exit_to_app
  </i>
</a>
`;
